### PR TITLE
Added spawning packets, and more framework

### PR DESCRIPTION
**Pull request by KodingDev**

### CODE REVIEW

File: src/main/kotlin/dev/cubxity/mc/protocol/packets/data/MagicRegistry.kt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 package dev.cubxity.mc.protocol.packets.data
 
 import dev.cubxity.mc.protocol.ProtocolVersion
-import dev.cubxity.mc.protocol.packets.data.enum.EnumGlobalEntityType
-import dev.cubxity.mc.protocol.packets.data.enum.EnumMobType
-import dev.cubxity.mc.protocol.packets.data.enum.EnumObjectType
+import dev.cubxity.mc.protocol.packets.data.enum.*
```
Comment (chachyyyy) [inline src/main/kotlin/dev/cubxity/mc/protocol/packets/data/MagicRegistry.kt]: wildcard!

File: src/main/kotlin/dev/cubxity/mc/protocol/net/NetOutput.kt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package dev.cubxity.mc.protocol.net
 
+import dev.cubxity.mc.protocol.entities.SimplePosition
 import io.netty.buffer.ByteBuf
 import java.util.*
```
Comment (chachyyyy) [inline src/main/kotlin/dev/cubxity/mc/protocol/net/NetOutput.kt]: wildcard!

File: src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt [hunk 3]
```diff
@@ -30,12 +26,6 @@ import java.lang.reflect.Constructor
 import java.util.*
```
Comment (chachyyyy) [inline src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt]: wildcard!

File: src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt [hunk 4]
```diff
@@ -4,11 +4,7 @@ import dev.cubxity.mc.protocol.net.PacketEncryption
 import dev.cubxity.mc.protocol.net.pipeline.TcpPacketCompression
 import dev.cubxity.mc.protocol.packets.Packet
 import dev.cubxity.mc.protocol.packets.PassthroughPacket
-import dev.cubxity.mc.protocol.packets.data.MagicRegistry
-import dev.cubxity.mc.protocol.packets.game.server.entity.spawn.SpawnExperienceOrbPacket
-import dev.cubxity.mc.protocol.packets.game.server.entity.spawn.SpawnGlobalEntityPacket
-import dev.cubxity.mc.protocol.packets.game.server.entity.spawn.SpawnMobPacket
-import dev.cubxity.mc.protocol.packets.game.server.entity.spawn.SpawnObjectPacket
+import dev.cubxity.mc.protocol.packets.game.server.entity.spawn.*
```
Comment (chachyyyy) [inline src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt]: wildcard!

Comment (asbyth) [reply src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt]: wildcard!

File: src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt [hunk 5]
```diff
@@ -36,6 +49,9 @@ class ProtocolSession @JvmOverloads constructor(
     val incomingVersion: ProtocolVersion = ProtocolVersion.V1_14_4,
     val outgoingVersion: ProtocolVersion = ProtocolVersion.V1_14_4
 ) : SimpleChannelInboundHandler<Packet>() {
+
+    private val objenesis = ObjenesisStd()
```
Comment (Cubxity) [inline src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt]: Not needed

File: src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt [hunk 6]
```diff
@@ -128,6 +144,19 @@ class ProtocolSession @JvmOverloads constructor(
     }
 
     fun defaultClientHandler() {
+        syncListeners += {
```
Comment (Cubxity) [inline src/main/kotlin/dev/cubxity/mc/protocol/ProtocolSession.kt]: Server doesn't send handshake to the client

**Review by Cubxity — CHANGES_REQUESTED** [review]

**Review by Cubxity — APPROVED** [review]

**Review by Cubxity — APPROVED** [review]

**Review by Cubxity — APPROVED** [review]

**Pull request merged**